util/src/openssl_crypto.cpp

```diff
@@ -31,30 +36,6 @@ using std::unique_ptr;
 // Deleter classes for using smart pointers that correctly manage objects from
 // the OpenSSL Crypto library, given OpenSSL Crypto objects use free functions
 // provided by the library rather than normal destructors.
-class BIGNUMDeleter {
- public:
-  void operator()(BIGNUM* num) const { BN_clear_free(num); }
-};
-class BNCTXDeleter {
- public:
-  void operator()(BN_CTX* obj) const { BN_CTX_free(obj); }
-};
-class ECKEYDeleter {
- public:
-  void operator()(EC_KEY* obj) const { EC_KEY_free(obj); }
-};
-class ECPOINTDeleter {
- public:
-  void operator()(EC_POINT* obj) const { EC_POINT_clear_free(obj); }
-};
-class EVPMDCTXDeleter {
- public:
-  void operator()(EVP_MD_CTX* obj) const { EVP_MD_CTX_free(obj); }
-};
-class EVPPKEYDeleter {
- public:
-  void operator()(EVP_PKEY* obj) const { EVP_PKEY_free(obj); }
-};
 class OPENSSLStringDeleter {
```

Comment:
Why do you still need this class?

Reply:
The `custom_deleter_unique_ptr` does not recognize the `OPENSSL_free` macro. It gives compilation error. I tried multiple times by trying to include header implementations, but didn't work. So, I kept it as is for the time being.

Reply:
you can wrap it in a function and then it'll work..